src/core/gps/qgsnmeaconnection.cpp

```diff
@@ -451,11 +453,20 @@ void QgsNmeaConnection::processGsvSentence( const char *data, int len )
       bool idAlreadyPresent = false;
       if ( mLastGPSInformation.satellitesInView.size() > NMEA_SATINPACK )
       {
-        for ( const QgsSatelliteInfo &existingSatInView : std::as_const( mLastGPSInformation.satellitesInView ) )
+        for ( QgsSatelliteInfo &existingSatInView : mLastGPSInformation.satellitesInView )
```

Comment:
I don't think this approach will work -- it's a qt thing due to implicit sharing of QList, but you're not allowed to modify the list directly while iterating over it using a range based loop.

For this to work you'd need to change to:

```suggestion
        for ( int i = 0; i < mLastGPSInformation.satellitesInView.size(); ++i )
        {
          QgsSatelliteInfo &existingSatInView = mLastGPSInformation.satellitesInView[i];
```